CRATE-STRUCTURE.md

```diff
@@ -0,0 +1,30 @@
+## Why so many crates?
```

Comment:
Nice document :)

Reply:
Haha thanks. Wasn't sure what to say, so feel free to add an modify!